userapp/views.py

```diff
@@ -19,8 +21,14 @@ class UserViewSet(viewsets.ModelViewSet,RetrieveModelMixin):
     def get_user_data(self, request):
         user = self.request.user
         serializer = UserSerializer(user)
-        return Response(serializer.data)
-
+        return Response(status = 200)
+    def get_queryset(self):
+        queryset = User.objects.all()
+        name = self.request.query_params.get('name')
+        if name is not None:
+            queryset = queryset.filter(first_name=name) or queryset.filter(last_name=name)
+        return queryset
```

Comment:
mozes i ovako :
if name is not None:
            return self.queryset.filter(Q(first_name=name) | Q(last_name=name))
return self.queryset